### PR TITLE
adds updates to branch 8_patchArticles

### PR DESCRIPTION
**Pull request by ericnabofa**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -1,227 +1,258 @@
...
+
+    const newVote = { inc_votes: 8 };
+
+    return request(app)
+      .patch("/api/articles/1")
+      .send(newVote)
+      .expect(200)
+      .then(({ body }) => {
+        const updatedArticle = body;
+        expect(updatedArticle).toMatchObject({
```
Comment (2202Hannah) [inline __tests__/app.test.js:246]: great use of toMatchObject 👍

File: app/app.model.js [hunk 2]
```diff
@@ -83,3 +83,16 @@ exports.insertCommentByArticle_Id = (article_id, username, body) => {
         return rows[0]
     })
 }
+
+exports.patchArticle = (inc_votes, article_id) => {
+    const queryString = `UPDATE articles 
+    SET votes = votes + $1 WHERE article_id = $2 
+    RETURNING *`
```
Comment (2202Hannah) [inline app/app.model.js:90]: nice :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -1,227 +1,258 @@
...
+          title: "Living in the shadow of a great man",
+          article_id: 1,
+          body: "I find this existence challenging",
+          topic: "mitch",
+          created_at: expect.any(String),
+          votes: existingArticle.votes + newVote.inc_votes,
+          article_img_url:
+            "https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+        });
+      });
```
Comment (2202Hannah) [inline __tests__/app.test.js:257]: You will need to consider error handling tests for your patch requests before you can merge to main

File: app/app.controller.js [hunk 4]
```diff
@@ -57,4 +57,13 @@ exports.postCommentByArticleId = (req, res, next) => {
         res.status(201).send({comment})
     })
     .catch(next)
+}
+
+exports.updateArticle = (req, res, next) => {
+    const {article_id} = req.params
+    const {inc_votes} =  req.body
+    patchArticle(inc_votes, article_id).then((updateArticle) => {
+        res.status(200).send(updateArticle)
```
Comment (2202Hannah) [inline app/app.controller.js]: We should send our response in an object on an appropriate key e.g. {updatedArticle : updatedArticle} which can be shortened to {updatedArticle}

**Review by 2202Hannah — CHANGES_REQUESTED** [review]
Hey Eric, really great start to task 8. You will also need to consider the following tests for error handling on your patch request:
- invalid article_id in the request
- article_id in the request doesn't exist in the database
- a bad request object with an incomplete body

Please submit another pr once you have added these tests. 

** there are a lot of tests for different endpoints included in this pr but I have only looked at the code related to `PATCH /api/articles/:article_id` during this code review.

File: __tests__/app.test.js (hunk 3 above)
Comment (ericnabofa) [reply __tests__/app.test.js:257]: yes..my bad
just did and doing another nchelp pr

Comment (ericnabofa) [reply __tests__/app.test.js:257]: done now

File: __tests__/app.test.js [hunk 5]
```diff
@@ -1,227 +1,301 @@
...
+        .send({ inc_votes: 5 })
         .expect(404)
         .then(({body}) => {
-          expect(body.msg).toBe('article does not exist');
+          expect(body.msg).toBe('article does not exist')
         });
     });
-});
-
+  });
```
Comment (Harrison365) [inline __tests__/app.test.js:301]: Nice changes 👍

**Comment by ericnabofa** [comment]
good to merge?

**Pull request merged**